tcms/testplans/static/testplans/js/get.js

```diff
@@ -192,3 +193,18 @@ function attachEvents(testCases, testPlanId, permissions) {
         return $(ev.target).closest('.js-testcase-row').data('testcase-pk');
     }
 }
+
+function toolbarEvents() {
+    $('.js-checkbox-toolbar').click(function(ev){
+        const isChecked = ev.target.checked,
+            testCaseRows = $('.js-testcase-row').find('input');
+
+        testCaseRows.each(function(index, tc) {
+            tc.checked = isChecked;
+        });
+
+        testCaseRows.click(function() {
+            ev.target.checked = false;
+        });
```

Comment:
I don't understand what this final block does. Explain.

Reply:
If a user select checkbox in the toolbar, all TC will be selected, after that if one of the TC is unchecked I want to uncheck checkbox in the toolbar. That will indicate that not all TC will be affected(if we do some action).
      The perfect case would be if on manual select we marked all checkbox the toolbar will be autoselected, but this is more complicated as we need counters and some additional logic.

Reply:
An elegant way, IMO, to implement what you want is to loop over all `testCaseRows` and `and` their `.checked` attributes together & assign the final result to `ev.target`.